lib/model/frames/form.js

```diff
@@ -144,11 +144,12 @@ Form.Extended = class extends Frame.define(
   'excelContentType',   readable,
   // counts of submissions in various review states
   'receivedCount',                      'hasIssuesCount',
-  'editedCount'
+  'editedCount',                        'hasDataset',  readable
```

Comment:
I like the new approach of `hasDataset` being part of the Form's extended metadata -- it's not too invasive of the existing Form infrastructure (e.g. migrating the entire form_defs table to add a new column) yet it is in an appropriate place for Frontend to access this info.